worker/metrics/collect/handler_test.go

```diff
@@ -198,11 +202,19 @@ func (c *mockConnection) SetDeadline(t time.Time) error {
 
 // Write implements the net.Conn interface.
 func (c *mockConnection) Write(data []byte) (int, error) {
+	c.Lock()
+	defer c.Unlock()
 	c.AddCall("Write", data)
 	c.data = data
 	return len(data), nil
 }
 
+func (c *mockConnection) readData() []byte {
+	c.Lock()
+	defer c.Unlock()
+	return c.data
```

Comment:
sorry, this isn't right, you need to return a _copy_ of the data, not just the slice header.